caffe2/operators/gru_unit_op.cc

```diff
@@ -35,6 +35,7 @@ value at X[t][n] >= seqLengths[n].
         "drop_states",
         "Bool to determine if hidden state is zeroes or passed "
         "along for timesteps past the given sequence_length.")
+    .Arg("no_sequence_lengths", "Ignore the sequence lengths input")
```

Comment:
usually, we do simple optional inputs like this one by changing number of inputs. I think, all other inputs are required. So you could always assume that if there are 3 inputs, they are prev, gates and t. 

Having a graph with 4 inputs but ignoring one of them is not good. It can, for example, confuse graph rewriting tools and gradient generators.

Reply:
so, in this case, I would prefer to keep consistency with LSTMUnit (you can see that I made almost character-for-character the same diff to LSTMUnit here https://github.com/caffe2/caffe2/pull/1764 - and in that case it was the only option), because these two features are already 90% copy-paste from each other, and gratuitously diverging them makes it harder to maintain.

If you feel strongly that it's better to diverge them, I'll make the change you've requested.